parlai/core/utils.py

```diff
@@ -713,6 +713,69 @@ def __contains__(self, key):
         """Determine if text contains any offensive words in the filter."""
         return self.contains_offensive_language(key)
 
+    def str_segment(self, text, dict_agent, max_length):
+        """
+        Function that segments a word without spaces into the most
+        probable phrase with spaces
+
+        :param string text: string to segment
+        :param DictionaryAgent dict_agent: Dictionary we use
+            to look at word frequencies
+        :param int max_length: max_length of string to segment
```

Comment:
maybe better:
`max_length: max length of a substring (word) in the string. default (None) uses the length of the string.